### DISCUSSION TITLE
Fallback for Static Path Matching

### DISCUSSION THREAD

**Discussion by jalcine** [trillium-rs/trillium#133] [Q&A] [answered]
Looking at the examples for `routefinder`, I'm noticing that I can't do something like `Router::new().get("/web/**/*", static_file_handler)`. [A more complete example from my project](https://git.jacky.wtf/indieweb/sele/src/branch/next/daemon/src/web/mod.rs#L98-L109):

```rust
        Router::new().get("/w/*", static_file_handler).get(
            "/",
            Box::new(|conn: trillium::Conn| async move {
                conn.with_header("location", "/w/").with_status(302).halt()
            }),
        )
```

My want to do that is to allow for complete URLs to be passed to the browser and for a fallback page for sorts to catch that and handle the client-side routing correctly (namely Svelte, in this case). The hope is that I can give it a specific path to `StaticFileHandler` but if none of the paths matched that I can tell it to use a specific path (that can potentially be checked at compile-time) to exist.

**Comment by jalcine** [comment]
What I've done in the interim is have a special query parameter that my CSR app can detect and do the client side redirection for.

In the CSR app:

```javascript
	onMount(() => {
                let goPath = $page.query.get('go') || (new URLSearchParams(location.search)).get('go');
		if (goPath) {
			goto(`/w/${goPath}`);
		}
	});
```

And in my app code:

```rust

            conn.with_status(Status::PermanentRedirect)
                .with_header("Location", format!("/w/?go=requests/{}", request.id))
                .halt()
```

**Comment by jbr** [comment, marked as answer]
I'm not certain I understand what you're looking for, but there are two ways to make a fall-through router match.  One is to mount a handler at `"*"`, and the other is to do `(Router::new().get(…), StaticCompiledHandler::new(…))` — the router will only halt the conn if it matches a route, so any unmatched route will continue onto the next handler. In routefinder, there's no distinction between "`**`" and `"*"` in that `"*"` matches anything including slashes, all the way to the end of the path

If instead, the need is to be able to always serve a specific file from the static file handler, there are changes I can release today that always serve a specific file regardless of path if instead of calling StaticFileHandler::new with a path to a directory, you call it with a single file

**Reply by jalcine** [reply]
The ability to always return a file if none matches would be excellent and I _think_ solve the use-case I have here.

P.S.: I don't know why I didn't get a notification for this.
